### PR TITLE
Feature/2017 01 13 kevin van den abeele node with typescript

### PR DESCRIPTION
**Pull request by beele**
Please proof read :)
Suggestions for improvements also welcome!

### CODE REVIEW

File: _posts/2017-01-12-Node-with-TypeScript.md [hunk 1]
```diff
@@ -0,0 +1,574 @@
...
+>NodeJS is a fantastic runtime to quickly and easily make projects. However as these projects tend to grow larger and larger, the shortcomings of JavaScript become more and more visible.
+This blog post will take a look at using TypeScript to write your Node code making it much more readable, introducing more OO like concepts whilst also making your code less error prone.
+
+### NodeJS and its use cases
+
+<p style="text-align: center;">
+  <img alt="NodeJS Powered by the V8 JavaScript engine" src="/img/node-with-typescript/V8-engine.jpg">
+</p>
+
+NodeJS has many use cases, it is an easy to pickup and use runtime. It uses google's V8 JavaScript engine to interpret and run JavaScript code.
```
Comment (tomverelst) [inline _posts/2017-01-12-Node-with-TypeScript.md]: Punctuation, grammar..
- "NodeJs has many use cases. It is easy.."
- Google's

File: _posts/2017-01-12-Node-with-TypeScript.md [hunk 2]
```diff
@@ -0,0 +1,574 @@
...
+The user does not have to worry about threading, that is taken care off by the runtime. You write your code and make use of the many asynchronous operations provided by Node.
```
Comment (tomverelst) [inline _posts/2017-01-12-Node-with-TypeScript.md]: Punctuation
"...about threading. This is taken care off..."

File: _posts/2017-01-12-Node-with-TypeScript.md [hunk 3]
```diff
@@ -0,0 +1,574 @@
...
+This will take care of any multithreading for you. However as you will read later in this blog post, making use of multiple Node instances to divide work is still possible!
```
Comment (tomverelst) [inline _posts/2017-01-12-Node-with-TypeScript.md]: However, ...

File: _posts/2017-01-12-Node-with-TypeScript.md [hunk 4]
```diff
@@ -0,0 +1,574 @@
...
+More on that later!
+<br/><br/>
+Node can be used for a variety of tasks:
+- Small yet efficient web server
+- Code playground, test something out quickly
+- Automation and tooling, instead of using ruby/python/...
+- IoT, Raspberry pi's and other devices that can run Node!
+
+<br/><br/>
+However you should not use node for computationally heavy tasks! While the V8 engine is highly performant, there are other much more performant options available for computationally heavy operations!
```
Comment (tomverelst) [inline _posts/2017-01-12-Node-with-TypeScript.md]: However, (comma) ..

File: _posts/2017-01-12-Node-with-TypeScript.md [hunk 5]
```diff
@@ -0,0 +1,574 @@
...
+- [The Node Package Manager](https://docs.npmjs.com/)
+- [Code school intro to NodeJS](https://www.codeschool.com/courses/real-time-web-with-node-js)
+
+### The old way, using plain JavaScript
+
+<p style="text-align: center;">
+  <img alt="JavaScript" src="/img/node-with-typescript/javascript.jpg">
+</p>
+
+Since NodeJS uses Google's V8 JavaScript engine, it speaks for itself that node interprets and runs regular JavaScript code. This has some pro and cons.
```
Comment (tomverelst) [inline _posts/2017-01-12-Node-with-TypeScript.md]: pro**s** and cons

File: _posts/2017-01-12-Node-with-TypeScript.md [hunk 6]
```diff
@@ -0,0 +1,574 @@
...
+While it is an easy language to pick up, it can be hard to master, javascript has always had some quirks and getting to know and how to avoid these can be tricky!
```
Comment (tomverelst) [inline _posts/2017-01-12-Node-with-TypeScript.md]: to master. Javascript ..

File: _posts/2017-01-12-Node-with-TypeScript.md [hunk 7]
```diff
@@ -0,0 +1,574 @@
...
+This can be handy to test something quickly.
+<br/><br/>
+It is also possible to run a JavaScript file directly. This can be done via:
+    {% highlight shell %}
+    
+    node path/to/javascript-file.js 
+       
+    {% endhighlight %}
+<br/>
+However most of the time you will not be using this way of running code. Most of the time you will use npm to install your dependencies and start the node instance:
```
Comment (tomverelst) [inline _posts/2017-01-12-Node-with-TypeScript.md]: However, (comma)

File: _posts/2017-01-12-Node-with-TypeScript.md [hunk 8]
```diff
@@ -0,0 +1,574 @@
...
+
+> Extensive documentation about the package.json file can be found on the [NPM website](https://docs.npmjs.com/files/package.json)
+
+### TypeScript you say!?
+
+<p style="text-align: center;">
+  <img alt="TypeScript" src="/img/node-with-typescript/typescript.jpg">
+</p>
+
+TypeScript has been around for some years now. It can be seen as a superset of JavaScript. It uses the same syntax but adds among other things compile time type checking.
```
Comment (tomverelst) [inline _posts/2017-01-12-Node-with-TypeScript.md]: "TypeScript can be seen as a superset.."

TypeScript **is** a superset

File: _posts/2017-01-12-Node-with-TypeScript.md [hunk 9]
```diff
@@ -0,0 +1,574 @@
...
+<p style="text-align: center;">
+  <img alt="TypeScript" src="/img/node-with-typescript/typescript.jpg">
+</p>
+
+TypeScript has been around for some years now. It can be seen as a superset of JavaScript. It uses the same syntax but adds among other things compile time type checking. 
+It also adds a more Object Oriented model. A detailed explenation of the differences of the prototype based JavaScript and a more Object Oriented language can be found on the 
+[Mozilla Developer website](https://developer.mozilla.org/nl/docs/Web/JavaScript/Guide/Details_of_the_Object_Model)
+<br/><br/>
+
+[TypeScript developed mainly by Microsoft](https://www.typescriptlang.org/) and is completely open source! This means developers can make suggestion and report bugs (and even fix these bugs if they want).
```
Comment (tomverelst) [inline _posts/2017-01-12-Node-with-TypeScript.md]: suggestion**s**

File: _posts/2017-01-12-Node-with-TypeScript.md [hunk 10]
```diff
@@ -0,0 +1,574 @@
...
+> TypeScript is a typed superset of JavaScript that compiles to plain JavaScript. Any browser. Any host. Any OS. Open source.
+
+<br/><br/>
+[TypeScript is very well documented](https://www.typescriptlang.org/docs/tutorial.html) and getting started with the language is fairly easy. 
+A lot of common development tools have support for TypeScript syntax checking. These include but are not limited to:
```
Comment (tomverelst) [inline _posts/2017-01-12-Node-with-TypeScript.md]: These include, (comma) ..

File: _posts/2017-01-12-Node-with-TypeScript.md [hunk 11]
```diff
@@ -0,0 +1,574 @@
...
+- Intellij
+- Webstorm
+- Atom
+- Visual Studio Code
+- ...
+
+<br/><br/>
+As Node applications regularly use other NPM dependencies it is required for the TypeScript compiler to know about these dependencies and what types they use.
+You could make or generate these typings yourself, however you can easily find these typings on [TypeSearch](https://microsoft.github.io/TypeSearch/) website.
```
Comment (tomverelst) [inline _posts/2017-01-12-Node-with-TypeScript.md]: You could make or generate these typings yourself.**However,** you can easily find these typings on

File: _posts/2017-01-12-Node-with-TypeScript.md [hunk 12]
```diff
@@ -0,0 +1,574 @@
...
+            this.arduino.init();
+        } else {
+            console.log('Skipping arduino setup, disabled in settings!');
+        }
+    };
+
+    {% endhighlight %}
+
+There are two Arduino implementations available. Both can execute a Scenario. 
+The first and simples implementation is the Johnny-Five Arduino implementation. This allows you to make use of the Johnny-Five framework to write dynamic code for the Arduino that can change at runtime.
```
Comment (tomverelst) [inline _posts/2017-01-12-Node-with-TypeScript.md]: simplest

File: _posts/2017-01-12-Node-with-TypeScript.md [hunk 13]
```diff
@@ -0,0 +1,574 @@
...
+authors: [kevin_van_den_abeele]
+title: 'Node with TypeScript'
+image: /img/node-with-typescript/node-ts.jpg
+tags: [Node, NodeJS, V8, JavaScript, TypeScript, IoT, Internet of Things]
+category: IoT
+comments: true
+---
+
+>NodeJS is a fantastic runtime to quickly and easily make projects. However as these projects tend to grow larger and larger, the shortcomings of JavaScript become more and more visible.
+This blog post will take a look at using TypeScript to write your Node code making it much more readable, introducing more OO like concepts whilst also making your code less error prone.
```
Comment (Turbots) [inline _posts/2017-01-12-Node-with-TypeScript.md]: to write your Node *application*

File: _posts/2017-01-12-Node-with-TypeScript.md [hunk 14]
```diff
@@ -0,0 +1,574 @@
...
+</p>
+
+NodeJS has many use cases, it is an easy to pickup and use runtime. It uses google's V8 JavaScript engine to interpret and run JavaScript code.
+The user does not have to worry about threading, that is taken care off by the runtime. You write your code and make use of the many asynchronous operations provided by Node.
+This will take care of any multithreading for you. However as you will read later in this blog post, making use of multiple Node instances to divide work is still possible!
+More on that later!
+<br/><br/>
+Node can be used for a variety of tasks:
+- Small yet efficient web server
+- Code playground, test something out quickly
```
Comment (Turbots) [inline _posts/2017-01-12-Node-with-TypeScript.md]: test something quickly (*to test out* bestaat niet)

**Comment by beele** [comment]
Thanks for the feedback, issues should be corrected (and now one line per sentence)

**Pull request merged**